.github/workflows/test-deploy.yml

```diff
@@ -102,47 +102,29 @@ jobs:
           registry-url: https://registry.npmjs.org
       - run: npm install
       - run:
-          npm ${{ inputs.release && 'publish' || 'publish --dry-run' }}
+          npm ${{ inputs.release && 'publish' || 'publish --dry-run' }} || ${{
+          inputs.release && 'true' || 'false' }}
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
-      - name: install ldid
-        run: |
+      - run: |
           sudo apt install --yes libplist-dev
           git clone --branch v2.1.5 git://git.saurik.com/ldid.git
           sudo g++ -pipe -o /usr/bin/ldid ldid/ldid.cpp -I. -x c ldid/{lookup2.c,sha1.h} -lplist -lcrypto
-      - id: build
-        name: build
-        run: |
-          echo ::set-output name=tag::${GITHUB_REF#refs/tags/}
+      - run: |
           cp node_modules/@npcz/magic/dist/magic.mgc assets/magic.mgc
           npx --yes pkg --no-bytecode --public-packages "*" --public package.json
-          rm assets/magic.mgc
-          for cmd in '' runner publish pr; do
-            build/cml-linux-x64 $cmd --version
-          done
+          for cmd in '' runner publish pr; do build/cml-linux-x64 $cmd --version; done
           cp build/cml-linux{-x64,}
           cp build/cml-macos{-x64,}
-      - uses: softprops/action-gh-release@v1
-        if: inputs.release
-        with:
-          files: |
-            build/cml-alpine-arm64
-            build/cml-alpine-x64
-            build/cml-linux-arm64
-            build/cml-linuxstatic-arm64
-            build/cml-linuxstatic-x64
-            build/cml-linux-x64
-            build/cml-linux
-            build/cml-macos-arm64
-            build/cml-macos-x64
-            build/cml-macos
-            build/cml-win-arm64.exe
-            build/cml-win-x64.exe
+      - if: inputs.release
+        run:
+          find build -type f | xargs gh release upload $(basename ${{
+          github.head_ref }})
         env:
           GITHUB_TOKEN: ${{ secrets.TEST_GITHUB_TOKEN }}
   images:
     needs: packages
     secrets: inherit
     uses: ./.github/workflows/images.yml
     with:
-      release: ${{ inputs.release }}
+      release: ${{ inputs.release || false }}
```

Comment:
why do we need this change?

Reply:
Because `inputs.release` might be unset, and cause the workflow to fail as @dacbd pointed out at e.g. [3172532632](https://github.com/iterative/cml/actions/runs/3172532632).